backend/homework/9/src/main/java/com/example/demo/repository/VehicleRepo.java

```diff
@@ -0,0 +1,67 @@
+package com.example.demo.repository;
+
+import com.example.demo.dto.VehicleRequestDTO;
+import com.example.demo.models.Vehicle;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@Repository
+public class VehicleRepo {
+    private final List<Vehicle> vehicles = new ArrayList<>();
+
+    public void createVehicle(Vehicle vehicle){
+        vehicles.add(vehicle);
+    }
+
+    public Vehicle getVehicleById(Long id){
+        for(Vehicle vehicle : vehicles){
+            if(Objects.equals(vehicle.getId(), id)){
+                return vehicle;
+            }
+        }
+        return null;
+    }
+
+    public Vehicle deleteVehicle(Long id){
+        for(Vehicle vehicle : vehicles){
+            if(Objects.equals(vehicle.getId(), id)){
+                vehicles.remove(vehicle);
+                return vehicle;
+            }
+        }
+        return null;
+    }
+
+    public Vehicle updateVehicle(Long id, VehicleRequestDTO vehicleRequestDTO){
+        for(Vehicle vehicle : vehicles){
+            if(Objects.equals(vehicle.getId(), id)){
+                vehicle.setName(vehicleRequestDTO.getName());
+                vehicle.setPrice(vehicleRequestDTO.getPrice()
+                        + vehicleRequestDTO.getTyre().getPrice() + vehicleRequestDTO.getSpeaker().getPrice());
+                vehicle.setFactory(vehicleRequestDTO.getFactory());
+                vehicle.setId(vehicleRequestDTO.getId());
+                vehicle.setSpeaker(vehicleRequestDTO.getSpeaker());
+                vehicle.setTyre(vehicleRequestDTO.getTyre());
+                vehicle.setId(vehicleRequestDTO.getId());
+
+                return vehicle;
+            }
+        }
+        return null;
+    }
+
+    public Vehicle mostExpVehicle(){
+        return vehicles.stream()
```

Comment:
Is it possible that the object will be null? Consider a handling mechanism for that scenario.